engine.py

```diff
@@ -196,7 +213,7 @@ def _define_qt_base(self):
         """
         check for pyside then pyqt
         """
-        
+
```

Comment:
Most of this logic is already handled by core. The only thing here really required is to check for qt availability, wrap qdialog and return the proxy if qt is not available. Seems like some work from the PySide support was left unmerged for [tk-shell](https://github.com/shotgunsoftware/tk-shell/pull/7/files).

I was super surprised that
a) this hadn't been done for tk-shotgun already by me
b) that it was never even merged in master for tk-shell.

Reply:
For the record, here is what I have in Photoshop engine: https://github.com/shotgunsoftware/tk-photoshopcc/blob/46256_pyside2_support/engine.py#L959
I still need to override the method for the QMessageBox patching, but all the work is done by tk-core.

Reply:
perfect! I'll modernize the shotgun engine to be on par with the new modern qt interfaces and i'll take a look at tk-shell at the same time!